### DISCUSSION TITLE
Range attribute in FromFactory

### DISCUSSION THREAD

**Discussion by Kralizek** [AutoFixture/AutoFixture#1352] [Q&A]
I need to create an instance of a type I don't have full control of and it has strict validation rules: it's a type representing a set of coordinates whose constructor accepts two variables of type double representing latitude (-90, 90) and longitude (-180, 180).

I was thinking about using one of `FromFactory` overloads and the `RangeAttribute` from `System.ComponentModel.DataAnnotations` but apparently the attribute constraint is ignored.

```csharp
fixture.Customize<GeoLocation>(o => o.FromFactory(([Range(-90, 90)] double latitude, [Range(-180, 180)] double longitude) => GeoLocation.TryCreate(latitude, longitude)));
```

Am I doing something wrong?

Here is a repro using DotNetFiddle: https://dotnetfiddle.net/N0bHCr

**Comment by aivascu** [comment]
I'm not sure but I think the RangeAttribute was supposed to work only on the test methods and on class properties which are being created by AutoFixture. I think you can try to create a proxy type that has members decorated with these attributes and use that object as a factory for your target type. Alternatively you might want to try to use the underlying ranged number request that AutoFixture uses to generate the bounded value.
Unfortunately I don't have a lot of time currently to research the solution. Perhaps over the weekend I can look into your question.

**Reply by Kralizek** [reply]
I ended up using a proxy type so don't worry, it's not urgent to me, nor blocking.

I opened this discussion just to make sure I wasn't making any obvious mistake and for future reference for other users of AF.

If this is an expected behavior and such supports comes in, it's all better. :)
